src/util/sharing_map.h

```diff
@@ -779,6 +791,73 @@ ::gather_all(const innert &n, delta_viewt &delta_view) const
   iterate(n, f);
 }
 
+SHARING_MAPT(void)::add_item_if_not_shared(
+  const innert &container,
+  const innert &inner,
+  delta_viewt &delta_view,
+  const bool only_common) const
+{
+  SM_ASSERT(!container.empty());
+  SM_ASSERT(!inner.empty());
+
+  std::stack<const innert *> stack;
+  stack.push(&inner);
+
+  do
+  {
+    const innert *ip = stack.top();
+    stack.pop();
+
+    SM_ASSERT(!ip->empty());
+
+    if(ip->is_internal())
+    {
+      const to_mapt &m = ip->get_to_map();
+      SM_ASSERT(!m.empty());
+
+      for(const auto &item : m)
+      {
+        const innert *i = &item.second;
+        stack.push(i);
+      }
+    }
+    else
+    {
+      SM_ASSERT(ip->is_container());
+
+      if(ip->shares_with(container))
+        return;
+
+      const leaft &l1 = container.get_container().front();
+
+      const leaf_listt &ll = ip->get_container();
+      SM_ASSERT(!ll.empty());
+
+      for(const auto &l : ll)
+      {
+        if(l1.shares_with(l))
+          return;
+
+        if(l1.get_key() == l.get_key())
+        {
+          // element is in both maps and not shared
+          delta_view.push_back(
+            {true, l1.get_key(), l1.get_value(), l.get_value()});
+          return;
```

Comment:
Again, I would have expected `continue` instead of returning.

Reply:
Similarly to above, since we have found a matching key, all other keys will be different and don't have to be included in the `delta_view`.